auth/auth.go

```diff
@@ -58,6 +58,22 @@ type Token struct {
 	Claims   map[string]interface{} `json:"-"`
 }
 
+func (t *Token) decodeFrom(s string) error {
+	claims := make(map[string]interface{})
+	if err := decode(s, &claims); err != nil {
+		return err
+	}
+	if err := decode(s, t); err != nil {
```

Comment:
I think it's worth adding a comment explaining why you need to decode twice.

Reply:
Done